src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -1515,14 +1516,15 @@ void InitAirLoops(EnergyPlusData &state, bool const FirstHVACIteration) // TRUE
         // store the results in AirToZoneNodeInfo()%CoolCtrlZoneNums and AirToZoneNodeInfo()%HeatCtrlZoneNums
 
         // Allocate scratch arrays for storing controlled zone numbers for each air loop.
-        state.dataSimAirServingZones->CtrlZoneNumsCool.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->CtrlZoneNumsHeat.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->ZoneInletNodesCool.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->ZoneInletNodesHeat.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->TermInletNodesCool.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->TermInletNodesHeat.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->TermUnitSizingNumsCool.allocate(state.dataGlobal->NumOfZones);
-        state.dataSimAirServingZones->TermUnitSizingNumsHeat.allocate(state.dataGlobal->NumOfZones);
+        int atuArraySize = max(state.dataGlobal->NumOfZones, state.dataDefineEquipment->AirDistUnit.size());
```

Comment:
Are all of these arrays always indexed by the same index? Or at least are all the ones that end in Heat or Cool always indexed by the same index? If so, suggest: 

```cpp
struct X { int CtrlZoneNum; int ZoneInletNode; int TermInletNode; int TermUnitSizingNum; } 
EPVector<struct X> XCool, XHeat;
```

I know that during the EnergyPlus 10X project I extolled the virtues of arrays of scalars over arrays of structs, but that was:
- For `struct`'s that had 200 fields, not 4. 
- For code in tight, high frequency loops that could potentially vectorize.

In general, I think EnergyPlus needs to swing the other way. Rather than a combination of arrays of giant `struct`'s and arrays of scalars, it would be better to have arrays of smaller, well-thought out `struct`'s. This would strike a good balance between array management and traversal overhead, cache/page locality, and code readability. This is exhibit A. </mini-rant>

Reply:
Not all of them are indexed the same. There might be groupings to consider. But these don't even belong in `state`, because they are only used in this function. Moving them out of `state` in the next pass here, along with localizing loop indices and a bunch of other integers.

Reply:
I am primarily using this as an opportunity to make a general point. I don't know whether that point applies on a case-by-case basis, and it may not apply here. I will probably make this point several more times in different places, where again it probably will not apply. 🤷

Reply:
structed as suggested in https://github.com/NREL/EnergyPlus/pull/9487/commits/7f8f512d79485d436441a8ca4a17cd0551291511